### ISSUE TITLE
Import problem with pygame

### ISSUE THREAD

**Issue by Redysz** [Nuitka/Nuitka#898]
Hello,
I'm not sure if it is a bug or a question. I tried to compile pygame project with nuitka and standalone option, always with the same error. Error:

```
Nuitka:INFO: Starting Python compilation.
Nuitka-Plugins:WARNING: Use '--plugin-enable=pylint-warnings' for: Understand PyLint/PyDev annotations for warnings.
Nuitka-Plugins:WARNING: Use '--plugin-enable=numpy' for: Numpy support for at least 'numpy'.
Traceback (most recent call last):
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\__main__.py", line 196, in <module>
    main()
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\__main__.py", line 189, in main
    MainControl.main()
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\MainControl.py", line 688, in main
    main_module = createNodeTree(filename=filename)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\MainControl.py", line 146, in createNodeTree
    Optimization.optimize(main_module.getOutputFilename())
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\optimizations\Optimization.py", line 625, in optimize
    makeOptimizationPass()
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\optimizations\Optimization.py", line 537, in makeOptimizationPass
    changed = optimizeModule(current_module)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\optimizations\Optimization.py", line 182, in optimizeModule
    optimizeShlibModule(module)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\optimizations\Optimization.py", line 177, in optimizeShlibModule
    Plugins.considerImplicitImports(module=module, signal_change=signalChange)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\plugins\Plugins.py", line 194, in considerImplicitImports
    plugin.considerImplicitImports(module, signal_change)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\plugins\PluginBase.py", line 165, in considerImplicitImports
    module_filename = self.locateModule(importing=module, module_name=full_name)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\plugins\PluginBase.py", line 445, in locateModule
    _module_package, module_filename, _finding = Importing.findModule(
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\importing\Importing.py", line 282, in findModule
    module_filename = _findModule(module_name=module_name)
  File "C:\Programy\Programowanie\Python\Python38\lib\site-packages\nuitka\importing\Importing.py", line 566, in _findModule
    assert not module_name.endswith("."), module_name
AssertionError: .
```

**python -m nuitka --version**
0.6.10
Python: 3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)]
Executable: C:\Programy\Programowanie\Python\Python38\python.exe
OS: Windows
Arch: x86_64

The way of execution:
python -m nuitka --standalone main.py (I've tried also with --follow-imports parameter)

Example of code:
```
import pygame
pygame.init()
pygame.font.init()

myfont = pygame.font.SysFont('Tahoma', 50)

screen = pygame.display.set_mode((800, 600))
clock = pygame.time.Clock()

BLACK = 0, 0, 0

running = True
while running:
    screen.fill(BLACK)
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    pygame.display.update()
    clock.tick(60)

pygame.quit()
```

**Comment by DarthNihilus1** [comment]
Having the same issue with pygame but non non distribution works

**Comment by Pro100rus32** [comment]
pip install -U "https://github.com/Nuitka/Nuitka/archive/factory.zip"

**Comment by kayhayen** [comment]
The standalone argument implies following imports, it makes no difference therefore. I will try to reproduce this, this seems to point to a bug in the importing logic. 

One thing I don't like about this code, that it doesn't tell us which top level source code triggered the issue. I will try to reproduce and make it more pinpointing to what import statement tripped it up.

**Comment by kayhayen** [comment]
That's better, with the error trace added, this is given:

> Nuitka-Plugins:INFO: implicit-imports: Problem locating <ModuleName pygame.mixer> implicit imports <ModuleName .>.

Seems our implicit import definitions are somehow buggy, although I am somehow suspecting, this is the `.pyi` parser that always can have challenges.

**Comment by kayhayen** [comment]
Yup, there is a line `from . import music as music` in that `mixer.pyi` file, making the parsing code claim that `.` is a dependency, where it should probably say `pygame.music`, which also doesn't exist as a Python module or extension class, I am suspecting it's a class.

**Comment by kayhayen** [comment]
This is fixed on the factory branch: https://nuitka.net/doc/factory.html

Your test program worked after making this fix. A workaround may also be to delete the line in question or maybe the whole `.pyi` file, but I am not entirely sure if that wouldn't make it miss some dependencies. I believe older versions of pygame didn't have this annotation.

Yours,
Kay

**Comment by kayhayen** [comment]
This is likely going to be hotfix material for 0.6.10.1 once it comes out.

**Comment by Redysz** [comment]
thank you for your quick and efficient response

**Comment by Redysz** [comment]
Ok, so it works (compiles) but when I'm trying to run it I get an error:

```
C:\Users\Karol Kurek\Desktop\gry\pygame_try1\main.dist>main.exe
pygame 2.0.0 (SDL 2.0.12, python 3.9.0)
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "C:\Users\KAROLK~1\Desktop\gry\PYGAME~2\MAIN~1.DIS\main.py", line 11, in <module>
FileNotFoundError: No such file or directory.
```

and when I copy main.py (source file) where main.exe is - it works, but I'm not sure if it should work in that way. I was sure that agfter compile with --standalone version it should work without python and source files.

**Comment by kayhayen** [comment]
You might want to check what `main.py` asked it to in line 11, don't you? @Redysz 

The example code didn't do that.

**Comment by Redysz** [comment]
Oh, my bad. You're right. There is image loading. Example code

```
import pygame
pygame.init()
pygame.font.init()

myfont = pygame.font.SysFont('Tahoma', 50)

screen = pygame.display.set_mode((800, 600))
clock = pygame.time.Clock()

BLACK = 0, 0, 0
cloud = pygame.image.load('image.png')

running = True
while running:
    screen.fill(BLACK)
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    pygame.display.update()
    clock.tick(60)

pygame.quit()
```

and I didn't have to copy main.py, I had to copy an image. So everything is ok.

**Comment by kayhayen** [comment]
Part of the latest pre-release.

**Comment by kayhayen** [comment]
Released as part of 0.6.11
